ansible/roles/ci/code_style_check/roslaunch_check/files/test_launchfiles.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# Maintained by devops@shadowrobot.com
+
+import sys
+import os
+import subprocess
+from xml.etree import ElementTree as et
+
+OUTPUT_PATH = "/home/user/testxml"
+
+
+def print_dictonary(dep_dict):
+    """Test function, may be removed later."""
+    error_count = 0
+    for key in dep_dict.keys():
+        print("Within the file {} include:".format(key))
+        for dep in dep_dict[key]:
+            print(dep)
+            error_count += 1
+        print()
+    print("Total errors found {}".format(error_count))
+    if error_count > 0:
+        exit(1)
+
+
+def strip_dependecies(dependency_list):
+    """Takes in a list of dependency and strips each string value."""
+    stripped_list = []
+    for dep in dependency_list:
+        stripped_list.append(dep.strip())
+    return stripped_list
+
+
+def gather_errors_from_produced_xmls():
+    """Goes through the testcase xmls and gathers all of the errors located. Places them in a dict."""
+    missing_dependencies = {"other_errors": []}
+    xml_files = [os.path.join(OUTPUT_PATH, f) for f in os.listdir(OUTPUT_PATH)]
+    for xml_file in xml_files:
+        tree = et.parse(xml_file)
+        root = tree.getroot()
+        testcase_name = root.find("testcase").attrib["name"]
+        for element in root.findall('system-out'):  # Use this incase there are many entries
+            missing_dependencies[testcase_name] = strip_dependecies(element.text.split("\n")[1:-1])
+    print_dictonary(missing_dependencies)
+
+
+def run_roslaunch_check(launch_files, test_dir):
+    """
+    Iterates through launch files and preforms roslaunch-check on them.
+    Puts produced xmls in a specified output folder.
+    """
+    for count, path_to_file in enumerate(launch_files):
+        file_name = path_to_file.split("/")[-1].split(".")[0]
+        path = os.path.join(test_dir, file_name + ".xml")
+        completed_process = subprocess.run(["rosrun", "roslaunch", "roslaunch-check", path_to_file, "-o", path])
```

Comment:
The issue with this is that it still produces a XML file when failing, as it returns 1 when it finds any error within a launch file, so by doing this it would only allow for one error to be checked per test. Maybe having a try and catch for extreme circumstances may work though